### PR TITLE
feat: update `with_metadata_operator.py` to support backwards compatibility with earlier versions than v2.5.0

### LINKED ISSUE #71: Backwards compatibility to version 2.4.3
# Goal

Use `airflow-provider-lakeFS` with Managed Airflow in Azure. Latest supported version of Airflow by Azure in Azure Data Factory is `v2.4.3`.

# Problem Description

Latest version of `airflow-provider-lakeFS` passes `TaskInstance.note` along with the metadata, making it only work with Apache Airflow v2.5.0 and later.

### PR DESCRIPTION
**Pull request by FredrikBakken**
Closes #71

### CODE REVIEW

File: lakefs_provider/operators/with_metadata_operator.py [hunk 1]
```diff
@@ -47,6 +48,9 @@ def _get_current_dag_dict(self, context: Dict[str, Any]) -> Dict[str, Any]:
         return ret
 
     def enrich_metadata(self, context: Dict[str, Any]):
+        # add 'note' value to metadata if Airflow is version 2.5.0 or later
+        if version.parse(airflow_version) >= version.parse("2.5.0"):
+            self.__metadata_templates["note"] = "{{dag_run.note}}"
         """Enrich metadata with values for lakeFS."""
```
Comment (arielshaqed) [inline lakefs_provider/operators/with_metadata_operator.py:54]: I think you misformatted the new code with respect to the PyDoc:

```suggestion
        """Enrich metadata with values for lakeFS."""
        # add 'note' value to metadata if Airflow is version 2.5.0 or later
        if version.parse(airflow_version) >= version.parse("2.5.0"):
            self.__metadata_templates["note"] = "{{dag_run.note}}"
```

**Review by arielshaqed — CHANGES_REQUESTED** [review]
Thanks!

Minor change but important: breaks PyDoc.

File: lakefs_provider/operators/with_metadata_operator.py (hunk 1 above)
Comment (arielshaqed) [reply lakefs_provider/operators/with_metadata_operator.py]: Another reason to love Python: docstring are _so_ fragile...

**Review by arielshaqed — APPROVED** [review]
Thanks!

**Comment by arielshaqed** [comment]
Yikes, provider test is failing 😭  

If you can give the Airflow version(s) on which you did manage to run this code, it might help debugging.

**Comment by FredrikBakken** [comment]
I was able to run this on Apache Airflow version 2.4.3, but I did not try it again on any later versions. I'll try to see if I can make it run on the latest version (2.6.3) locally too, or if I encounter any problems. Reporting back to you soon!

**Comment by FredrikBakken** [comment]
I've now executed both the `LakeFSCommitOperator` and the `LakeFSMergeOperator` on a local Airflow instance, where both versions `2.4.3` and `2.6.3` have been able to run without any problems. However, I have not triggered a case where `note` is used and this field is therefore not showing in the LakeFS UI.

**Comment by arielshaqed** [comment]
Not sure why the provider test consistently fails.  This is usually quite serious; I'll have time to look at it tomorrow.

**Pull request merged**